modules/core/src/v2/coins/xlm.ts

```diff
@@ -759,4 +769,125 @@ export class Xlm extends BaseCoin {
+      999999,
+      parseInt(derivationPathInput.slice(0, 7), 16),
+      parseInt(derivationPathInput.slice(7, 14), 16),
+    ];
+    const derivationPath = 'm/' + derivationPathParts
+      .map((part) => `${part}'`)
+      .join('/');
+    const derivedKey = XlmKeyDeriver.derivePath(derivationPath, key).key;
+    const keypair = stellar.Keypair.fromRawEd25519Seed(derivedKey);
+    return {
+      key: keypair.publicKey(),
+      derivationPath,
+    };
+  }
+}
+
+export interface XlmKeypair { key: Buffer, chainCode: Buffer }
+
+export class XlmKeyDeriver {
+  /**
+   * This key derivation code was copied and adapted from:
+   * https://github.com/chatch/stellar-hd-wallet/blob/612c12325ca9047dce460016fb7d148f55f575ca/src/hd-key.js
+   *
+   * There have been some slight modifications to improve typescript support.
+   *
+   * The original ed25519-hd-key module is licensed under "GPL-3".
+   */
+
+  private static readonly ED25519_CURVE = 'ed25519 seed';
+  private static readonly HARDENED_OFFSET = 0x80000000;
+  private static readonly PATH_REGEX = new RegExp("^m(\\/[0-9]+')+$");
+
+
+  /**
+   * Derive a SLIP-0010 key given a path and master key seed.
+   *
+   * https://github.com/satoshilabs/slips/blob/master/slip-0010.md
+   * https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+   * https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0005.md
+   *
+   * @param path derivation path
+   * @param seed key seed
+   */
+  public static derivePath(path: string, seed: string): XlmKeypair {
+    if (!this.isValidPath(path)) {
+      throw new Error('Invalid derivation path');
+    }
+    const { key, chainCode } = this.getMasterKeyFromSeed(seed);
+    const segments = path
+      .split('/')
+      .slice(1)
+      .map(this.replaceDerive);
+    return segments.reduce(
+      (parentKeys, segment) =>
+        this.CKDPriv(parentKeys, segment + XlmKeyDeriver.HARDENED_OFFSET),
+      { key, chainCode }
+    );
+  }
+
+  /**
+   * Generate a SLIP-0010 master key from the entropy seed
+   *
+   * @param seed master key seed used to recreate master key
+   */
+  private static getMasterKeyFromSeed(seed: string) {
+    const hmac = createHmac('sha512', XlmKeyDeriver.ED25519_CURVE);
+    const I = hmac.update(Buffer.from(seed, 'hex')).digest();
+    const IL = I.slice(0, 32);
+    const IR = I.slice(32);
+    return {
+      key: IL,
+      chainCode: IR,
+    };
+  }
+
+  /**
+   * Calculate a child private key given the parent key, the chain code, and the child index.
+   *
+   * @param key parent key
+   * @param chainCode chain code for parent key
+   * @param index index of child to derive
+   */
+  private static CKDPriv({ key, chainCode }: XlmKeypair, index: number): XlmKeypair {
+    const indexBuffer = Buffer.allocUnsafe(4);
+    indexBuffer.writeUInt32BE(index, 0);
+    const data = Buffer.concat([Buffer.alloc(1, 0), key, indexBuffer]);
+    const I = createHmac('sha512', chainCode)
```

Comment:
Are `I`, `IL`, and `IR` naming conventions specific to this lib or XLM or this curve that I'm not familiar with? If so, and you feel other contributors will pick this up, pls proceed. Otherwise, we may want to create more meaningful const names for them.

Reply:
They are variable names from the algorithm used for key derivation in BIP-32 and SLIP-0010.